src/components/PageDefinitions.js

```diff
@@ -279,12 +280,14 @@ export class PageDefinitions extends AbstractPageDefinitions {
     }
   }
 
-  onTextDrop = url => {
+  onTextDrop = content => {
     const { dispatch } = this.props
-    const path = EntitySpec.fromUrl(url)
-
-    if (path.errors) dispatch(uiNotificationNew({ type: 'warning', message: path.errors, timeout: 5000 }))
-    else this.onAddComponent(path)
+    if (!isJson(content)) {
```

Comment:
if the drop had `type === 'application/json'` could also be dispatching based on that. may still be handy to allow dropping of arbitrary "text" that happens to be json...